wayne_full_commission and wayne_building_authority - 457 wayne commission aggregate

novellac opened the pull request:
Spider and test for wayne_full_commission (https://www.waynecounty.com/elected/commission/full-commission.aspx)

@diaholliday can you take a look at the description in the spider and compare it to the descriptive text on the website?

References #456

novellac commented:
Added the wayne_building_authority spider. Feedback welcome.

diaholliday commented:
This description from the home page is better—can you replace with the text below? 

"The Wayne County Commission and its employees are the legislative branch of county government. The chief role of the commission is to adopt a budget and enact ordinances. The commission also approves contracts, appointments and rules. The money is spent and ordinances are enforced through the administrative branch."

novellac commented:
Description updated - ready for review

pjsier commented:
Should we leave the description blank here and include that text in the agency description outside of the scrapers?

mkrump commented on city_scrapers/spiders/wayne_building_authority.py:
```diff
@@ -0,0 +1,78 @@
+        # There are several tables with old meetings. Look for the current year
+        # and get all rows which have children that contain text.
+        current_year = datetime.now().year
+        return response.xpath('//section[contains(.,"%s")]//tbody/tr[child::td/text()]' %current_year)
```
Could maybe replace the comment and use a descriptive name for the xpath? 

```
current_year_non_empty_rows = '//section[contains(.,"%s")]//tbody/tr[child::td/text()]'
```

mkrump commented on city_scrapers/spiders/wayne_building_authority.py:
```diff
@@ -0,0 +1,78 @@
+        """
+
+        # Our status may be buried inside a number of other elements
+        status_str = item.xpath('.//td[4]/*/text() | .//td[4]/*/*/text() | .//td[4]/text()').extract_first()
```
Would it be possible to replace `'.//td[4]/*/text() | .//td[4]/*/*/text() | .//td[4]/text()'` with just `.//td[4]//text()` and rely on `//` to get all text below the `td[4]` node? Could join text together if possibility for multiple text nodes.

mkrump commented on city_scrapers/spiders/wayne_building_authority.py:
```diff
@@ -0,0 +1,78 @@
+        # Our status may be buried inside a number of other elements
+        status_str = item.xpath('.//td[4]/*/text() | .//td[4]/*/*/text() | .//td[4]/text()').extract_first()
+        # Meetings that are truly cancelled will be marked here.
+        if re.search(r'cancel', status_str, re.IGNORECASE):
```
could just check for 'cancel' in `status_str` (e.g. `if 'CANCEL' in status_str.upper()`), or I think could instead pass status string to `self._generate_status(data, status_str)`.

novellac commented:
Thanks for the review - I reworked a couple of the things you mentioned.

novellac commented:
@pjsier Can you review (and merge if appropriate)?

pjsier commented on city_scrapers/spiders/wayne_building_authority.py:
```diff
@@ -0,0 +1,78 @@
+        """
+        Event description
+        """
+        desc = ("The County of Wayne will provide necessary reasonable "
```
Based on the conversation today it would be good to just return an empty string here, which can probably just be done in the returned dict without using the `_parse_description` method

pjsier commented on city_scrapers/spiders/wayne_full_commission.py:
```diff
@@ -0,0 +1,37 @@
+        @TODO Event description should be based on feedback from CB, since
+        it also contains alternate location info.
+        """
+        desc = ("The Wayne County Commission and its employees are the "
```
Same as the previous comment about the descriptions, we can just include an empty string here

novellac commented:
I think I got everything? As per the discussion today, I removed the description method from spiders and added a default empty description into the mixin.

pjsier commented:
It looks like that covers the description, but could you make the updates for simplifying the one xpath to `.//td[4]//text()` (if it works) and doing a substring check instead of regex for the "cancel" string in `wayne_building_authority`?

novellac commented:
Ah perhaps I just did it for the mixin. _parse_status no longer needed in the wayne_building_authority spider.

pjsier commented:
Those changes look good to me! Squashing and merging now
